cJSON.c

```diff
@@ -2548,7 +2548,12 @@ CJSON_PUBLIC(cJSON *) cJSON_CreateIntArray(const int *numbers, int count)
     }
 
     a = cJSON_CreateArray();
-    for(i = 0; a && (i < (size_t)count); i++)
```

Comment:
when allocation failure, `cJSON_CreateArray()` will return NULL, then it won't enter the for loop.

Reply:
yes, it won't enter the for loop, but 'a->child' will crush.

Reply:
the line after for loop, a->child->prev = n;

Reply:
thanks, I didn't notice that.